ckanext/datagovuk/tests/test_s3_resources_plugin.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+import cgi
+from StringIO import StringIO
+
+import botocore.session
+import pylons.config as config
+from botocore.stub import ANY, Stubber
+from ckanext.datagovuk.plugin import DatagovukPlugin
+from mock import patch
+from nose.tools import assert_raises, assert_raises_regexp
+
+
+class MockAction:
+    pkg = None
+    rsc = None
+
+    def __init__(self, _type):
+        self._type = _type
+
+    def __call__(self, *arg):
+        if self._type == "package_show":
+            return self.pkg or {"name": "test"}
+        elif self._type == "resource_show":
+            return self.rsc or {
+                "url_type": "test",
+                "name": "resource-id",
+                "url": "http://test-resource",
+            }
+
+
+@patch.dict(
+    "pylons.config",
+    {
+        "ckan.datagovuk.s3_aws_access_key_id": "key-id",
+        "ckan.datagovuk.s3_aws_secret_access_key": "secret-key",
+        "ckan.datagovuk.s3_bucket_name": "test-bucket",
+        "ckan.datagovuk.s3_url_prefix": "https://s3.amazonaws.com/test/",
+        "ckan.datagovuk.s3_aws_region_name": "eu-west-1",
+    },
+)
+class TestS3ResourcesPlugin:
+    upload = cgi.FieldStorage()
+    upload.file = StringIO("hello world")
+    resource = {
+        "package_id": u"some-pointless-data-3",
+        "clear_upload": u"",
+        "url": "organogram-senior.csv",
+        "timestamp": "2020-01-09T12-11-41Z",
+        "datafile-date": u"",
+        "upload": upload,
+        "name": "2020-01-09 Organogram (Senior)",
+    }
+
+    @patch.dict("pylons.config", {}, clear=True)
+    def test_s3_config_exception(self):
+        plugin = DatagovukPlugin()
+        with assert_raises_regexp(Exception, "Required S3 config options missing"):
```

Comment:
Do we need to use `assert_raises_regexp`, will `assert_raises` work?

Reply:
I used `assert_raises_regexp` because the code raises a plain `Exception` with a message and I wanted to ensure it was the specific exception being raised. The alternative would be to raise a more specific exception, maybe `NameError`?